tests/destructor/t5342.nim

```diff
@@ -0,0 +1,23 @@
+discard """
+  matrix: "--gc:refc; --gc:arc"
+  output: '''
+1
+2
+here
+2
+1
+'''
+"""
+
+
```

Comment:
```nim
when true:
  type
    A = object
      id: int
    B = object
      a: A
  proc `=destroy`(a: var A) = echo a.id
  template main =
    var x = A(id: 1)
    var y = B(a: A(id: 2))
    `=destroy`(x)
    `=destroy`(y)
    echo "here"
  static: main()
  main()
```

* fails in VM
* fails in RT js

is that tracked somewhere?

Reply:
Open a new issue?

Reply:
=> filed https://github.com/nim-lang/Nim/issues/17237